### DISCUSSION TITLE
Set field to null via API

### DISCUSSION THREAD

**Discussion by RobinBeismann** [netbox-community/netbox#6600] [Q&A] [answered]
Hi guys,

I'm trying to set a field (primary_ip and primary_ip4 in this case) to null via the REST API, basically removing its value. This is possible via the UI but I can't figure out how to do it via the API.
If I post NULL, null or NULL() to the primary_ip(4) field, then Netbox doesn't seem to be processing the value I push..

Anyone got an advice on that one?

Thanks,
Robin

**Comment by candlerb** [comment, marked as answer]
You'll want [PUT or PATCH](https://netbox.readthedocs.io/en/stable/rest-api/overview/#what-is-a-rest-api) rather than POST.  Can you show the code or curl command line you're trying?

**Reply by RobinBeismann** [reply]
I've tried patch as I use for all other updates, basically it's just Powershell with 
```
Invoke-RestMethod -Method "PATCH" -URI "https://netbox/api/virtualization/virtual-machine/id/ -Body (@{ primary_ip = null()} | ConvertTo-JSON ) -Headers [...]
```
However this does not remove the primary_ip.

The code might be a bit different, already shut down the PC and did it out of my head.

**Reply by candlerb** [reply]
Can you get that Microsoft command to show the actual raw body that it sends?  And did you send Content-Type: application/json ?

**Reply by RobinBeismann** [reply]
Never mind, found my issue..

I was able to update anything except that one, turns out I missed the @ in front of my hashtable so I sent a Scriptblock instead of a stringified JSON.

Thanks for the help through!
